### ISSUE TITLE
[HttpKernel] new catchExceptions() method in the client class

### ISSUE THREAD

**Issue by xabbuh** [symfony/symfony-docs#8427]
The method allows to turn catching exceptions off which makes them visible in functional tests (see symfony/symfony#22890).

**Comment by javiereguiluz** [comment]
Can anyone please provide a super simple code example using this feature in action? Maybe @kbond who implemented this feature originally? Thanks!

**Comment by xabbuh** [comment]
The following example should work:

```php
public function testSomething()
{
    $client = static::createClient();
    $client->catchExceptions(false);

    // ...
}
```

**Comment by javiereguiluz** [comment]
Fixed by #10098.
